recipes-omnect/initrdscripts/omnect-os-initramfs/grub-sh

```diff
@@ -19,19 +19,49 @@ get_bootloader_env_var()
     test_and_mount_boot || return 1
     local value=$(grub-editenv ${ROOTFS_DIR}/boot/EFI/BOOT/grubenv list | grep ${var_name}\= | cut -d'=' -f2)
     debug "get_bootloader_env_var ${var_name}=${value}"
-    echo ${value}
+    echo "${value}"
 }
 
 # set/delete bootloader environment variable
 set_bootloader_env_var()
 {
     local  var_name="$1"
     local var_value="$2"  # empty string to delete variable
+    msg "set_bootloader_env_var" "${var_name}" "${var_value}"
 
     test_and_mount_boot || return 1
     if [ -n "$var_value" ];then
-        run_cmd grub-editenv ${ROOTFS_DIR}/boot/EFI/BOOT/grubenv set ${var_name}=${var_value}
+        grub-editenv ${ROOTFS_DIR}/boot/EFI/BOOT/grubenv set "${var_name}"="${var_value}" || return 1
     else
-        run_cmd grub-editenv ${ROOTFS_DIR}/boot/EFI/BOOT/grubenv unset ${var_name}
+        grub-editenv ${ROOTFS_DIR}/boot/EFI/BOOT/grubenv unset "${var_name}" || return 1
```

Comment:
run_cmd left out intentionally?

Reply:
yes, depending on the content of value `run_cmd` will fail due to escape problems